burn-core/src/nn/lstm/lstm.rs

```diff
@@ -0,0 +1,322 @@
+use burn_tensor::activation;
+
+use crate as burn;
+
+use crate::config::Config;
+use crate::module::Module;
+use crate::nn::lstm::gate_controller;
+use crate::nn::Initializer;
+use crate::nn::LinearConfig;
+use crate::tensor::backend::Backend;
+use crate::tensor::Tensor;
+
+use super::gate_controller::GateController;
+
+#[derive(Config)]
+pub struct LSTMConfig {
+    /// The size of the input features.
+    pub d_input: usize,
+    /// The size of the hidden state.
+    pub d_hidden: usize,
+    /// If a bias should be applied during the Lstm transformation.
+    pub bias: bool,
+    /// Lstm initializer
+    /// TODO: Make default Xavier initialization, which should be
+    /// a better choice. https://github.com/burn-rs/burn/issues/371
+    #[config(default = "Initializer::Uniform(0.0, 1.0)")]
+    pub initializer: Initializer,
+    /// The batch size
+    pub batch_size: usize,
+}
+
+/// The Lstm module. This implementation is for a unidirectional, stateless, Lstm.
+#[derive(Module, Debug)]
+pub struct Lstm<B: Backend> {
+    input_gate: GateController<B>,
+    forget_gate: GateController<B>,
+    output_gate: GateController<B>,
+    cell_gate: GateController<B>,
+    batch_size: usize,
+    d_hidden: usize,
+}
+
+impl LSTMConfig {
```

Comment:
Can you we also make the name the Camel, like LstmConfig? Someone pointed out naming is not up to convention: https://rust-lang.github.io/api-guidelines/naming.html